static/js/recent_topics.js

```diff
@@ -40,6 +42,16 @@ exports.process_message = function (msg) {
     if (topic.last_msg.timestamp <= msg.timestamp) {
         topic.last_msg = reduce_message(msg);
     }
+
+    // Maintain sender_ids in order of they send msgs
+
+    const sender = msg.sender_id;
+    // Remove sender if already exists
+    if (topic.senders.indexOf(sender) !== -1) {
+        topic.senders.splice(topic.senders.indexOf(sender), 1);
+    }
+    topic.senders.push(sender);
```

Comment:
I think these data are already available in `static/js/recent_senders.js`, which we use for typeahead; so it might be better to just grab the data from there rather than having this module track that data in a duplicate fashion.

(This will be particularly useful if in the future we start having the server send data about recent senders to clients).

Reply:
`recent_senders` doesn't contain the last msg id that a user has sent. It contains the first msg id user sends in the topic. This means if I sort them, I won't get them in the order of their the last msg sent. So, I am tracking them here. Moved to code 955d2c317b for better representation. First 6 commits should be good to merge, the rest is UI part. Opened #14603 for just the data processing commits.

Reply:
You were right about recent senders. I mistook the msg_ids for something else at that time.